Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -791,55 +791,73 @@ public boolean chooseTargetAmount(Outcome outcome, TargetAmount target, Ability
             abilityControllerId = target.getAbilityController();
         }
 
-        while (canRespond()) {
-            Set<UUID> possibleTargets = target.possibleTargets(source == null ? null : source.getSourceId(), abilityControllerId, game);
-            boolean required = target.isRequired(source != null ? source.getSourceId() : null, game);
+        int amount = target.getAmountRemaining();
+        if (amount < 1) {
+            return false;
+        }
+
+        Map<UUID, String> targets = new LinkedHashMap<>(amount);
+        Set<UUID> possibleTargets = target.possibleTargets(source == null ? null : source.getSourceId(), abilityControllerId, game);
```

Comment:
Same for `possbileTargets` (it must updated from actual target, see comment about targets above ).

Also you are removes selected targets from possible list. I don't test it but it can be a little UX problem. Possible targets list uses by client to highlighting the cards (yellow color), so after removing the target user can't see it as selectable. But old choose dialog logic allows to select already selected target and remove selection or change amount. So users must see that targets can be selected/changed.